src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -400,12 +403,12 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
 
   /** Reset company type values to original. */
   resetType () {
-    this.setEntityType(this.getEntitySnapshot?.businessInfo?.legalType)
+    this.setEntityType(this.getOriginalBusinessInfo?.legalType || null)
```

Comment:
This is to make sure we don't save undefined props in the store (which are not reactive).